specs/_features/eip7732/fork-choice.md

```diff
@@ -260,185 +262,151 @@ def get_checkpoint_block(store: Store, root: Root, epoch: Epoch) -> Root:
 ### `is_supporting_vote`
 
 ```python
-def is_supporting_vote(store: Store, node: ChildNode, message: LatestMessage) -> bool:
+def is_supporting_vote(store: Store, node: ForkChoiceNode, message: LatestMessage) -> bool:
     """
     Returns whether a vote for ``message.root`` supports the chain containing the beacon block ``node.root`` with the
-    payload contents indicated by ``node.is_payload_present`` as head during slot ``node.slot``.
+    payload contents indicated by ``node.payload_status`` as head during slot `node`'s slot.
     """
+    block = store.blocks[node.root]
     if node.root == message.root:
-        # an attestation for a given root always counts for that root regardless if full or empty
-        # as long as the attestation happened after the requested slot.
-        return node.slot <= message.slot
-    message_block = store.blocks[message.root]
-    if node.slot >= message_block.slot:
-        return False
-    ancestor = get_ancestor(store, message.root, node.slot)
-    return (node.root == ancestor.root) and (node.is_payload_present == ancestor.is_payload_present)
+        if node.payload_status == PAYLOAD_STATUS_PENDING:
+            return True
+        if message.slot <= block.slot:
```

Comment:
This one is tricky to read. The case `message.slot == block.slot` should have been handled by the previous `if` statement, perhaps a comment here would help.